pkg/controller/direct/bigquerydataset/bigquerydataset_mappings.go

```diff
@@ -21,210 +21,309 @@
+	out.DefaultTableExpiration = time.Duration(direct.ValueOf(in.DefaultTableExpirationMs)) * time.Millisecond
+	out.DefaultEncryptionConfig = EncryptionConfiguration_ToProto(mapCtx, in.DefaultEncryptionConfiguration)
 	out.Description = direct.ValueOf(in.Description)
-	out.FriendlyName = direct.ValueOf(in.FriendlyName)
-	out.DatasetReference = DatasetReference_ToAPI(mapCtx, in, name)
+	out.ExternalDatasetReference = DatasetSpec_ToExternalDatasetReference(mapCtx, in)
 	out.Location = direct.ValueOf(in.Location)
-	out.IsCaseInsensitive = direct.ValueOf(in.IsCaseInsensitive)
 	if in.MaxTimeTravelHours != nil {
-		out.MaxTimeTravelHours, _ = strconv.ParseInt(direct.ValueOf(in.MaxTimeTravelHours), 10, 64)
+		maxHours, _ := strconv.ParseInt(direct.ValueOf(in.MaxTimeTravelHours), 10, 64)
+		out.MaxTimeTravel = time.Duration(maxHours) * time.Hour
 	}
+	out.Name = direct.ValueOf(in.FriendlyName)
 	out.StorageBillingModel = direct.ValueOf(in.StorageBillingModel)
 	return out
 }
-func BigQueryDatasetSpec_FromAPI(mapCtx *direct.MapContext, in *api.Dataset) *krm.BigQueryDatasetSpec {
+func BigQueryDatasetSpec_FromProtoMetadata(mapCtx *direct.MapContext, in *pb.DatasetMetadata) *krm.BigQueryDatasetSpec {
 	if in == nil {
 		return nil
 	}
 	out := &krm.BigQueryDatasetSpec{}
 	accessList := []krm.Access{}
 	for _, access := range in.Access {
-		curAccess := Access_FromAPI(mapCtx, access)
+		curAccess := AccessEntry_FromProto(mapCtx, access)
 		accessList = append(accessList, direct.ValueOf(curAccess))
 	}
 	out.Access = accessList
 	out.DefaultCollation = direct.LazyPtr(in.DefaultCollation)
-	out.DefaultPartitionExpirationMs = direct.LazyPtr(in.DefaultPartitionExpirationMs)
-	out.DefaultTableExpirationMs = direct.LazyPtr(in.DefaultTableExpirationMs)
-	out.DefaultEncryptionConfiguration = EncryptionConfiguration_FromAPI(mapCtx, in.DefaultEncryptionConfiguration)
+	out.DefaultPartitionExpirationMs = direct.LazyPtr(in.DefaultPartitionExpiration.Milliseconds())
+	out.DefaultTableExpirationMs = direct.LazyPtr(in.DefaultTableExpiration.Milliseconds())
+	out.DefaultEncryptionConfiguration = EncryptionConfiguration_FromProto(mapCtx, in.DefaultEncryptionConfig)
 	out.Description = direct.LazyPtr(in.Description)
-	out.FriendlyName = direct.LazyPtr(in.FriendlyName)
+	out.FriendlyName = direct.LazyPtr(in.Name)
 	out.Location = direct.LazyPtr(in.Location)
-	out.IsCaseInsensitive = direct.LazyPtr(in.IsCaseInsensitive)
-	maxTime := strconv.FormatInt(in.MaxTimeTravelHours, 10)
-	out.MaxTimeTravelHours = direct.LazyPtr(maxTime)
+	maxTimeInHours := fmt.Sprintf("%v", in.MaxTimeTravel.Hours())
+	out.MaxTimeTravelHours = &maxTimeInHours
 	out.StorageBillingModel = direct.LazyPtr(in.StorageBillingModel)
 	return out
 }
-func BigQueryDatasetStatus_FromAPI(mapCtx *direct.MapContext, in *api.Dataset) *krm.BigQueryDatasetStatus {
+func BigQueryDatasetStatus_FromProtoMetadata(mapCtx *direct.MapContext, in *pb.DatasetMetadata) *krm.BigQueryDatasetStatus {
```

Comment:
nit: rename to `BigQueryDatasetStatus_FromProtoMetadata`. I understand the intention, but I don't think we have a pattern of suffixing the mapping function with part of the proto message name.